.pipelines/ci.yml

```diff
@@ -20,97 +20,60 @@ pr:
-    variables:
-      HOME: $(Agent.BuildDirectory)
-
-    steps:
-      - template: ./templates/template-checkout.yml
-      - script: |
-          set -xe
-          make test-python
-          [[ -z "$(git status -s)" ]]
-        displayName: 🧪Run Python Unit Tests
-        target: python
-
-  - job: Golang_Unit_Tests
-    pool:
-      name: 1es-aro-ci-pool
-    variables:
-      GOCACHE: /tmp/gocache
-    steps:
-      - template: ./templates/template-checkout.yml
-
-      - script: |
-          set -xe
-          go version
-          go env
-        displayName: Print Go version & env
-        target: golang
-
-      - script: |
-          set -xe
-          make generate
-          [[ -z "$(git status -s)" ]]
-        displayName: ⚙️ Run Golang code generate
-        target: golang
-
-      - script: |
-          set -xe
-          make build-all
-          [[ -z "$(git status -s)" ]]
-        displayName: 🕵️ Build Golang code
-        target: golang
-
-      - script: |
-          set -xe
-          make unit-test-go
-        displayName: 🧪 Run Golang unit tests
-        target: golang
 
-      - task: PublishTestResults@2
-        displayName: 📊 Publish tests results
-        inputs:
-          testResultsFiles: $(System.DefaultWorkingDirectory)/**/report.xml
-        condition: succeededOrFailed()
+stages:
+- stage: NEW_HOTNESS
+  jobs:
+    - job: Build_Lint_Test_RP_And_Portal
+      pool:
+        name: 1es-aro-ci-pool
+      steps:
+        - template: ./templates/template-checkout.yml
+        - script: |
+            set -xe
+            export NO_CACHE=true
+            make ci-rp
 
-      - script: |
-          set -xe
-          go run github.com/axw/gocov/gocov@v1.1.0 convert cover.out | go run github.com/AlekSi/gocov-xml@v1.1.0 > coverage.xml
-        displayName: ⚙️ Process Reports
-        condition: succeededOrFailed()
-        target: golang
+    - job: Build_and_Test_Az_ARO_Extension
+      pool:
+        name: 1es-aro-ci-pool
+      steps:
+        - template: ./templates/template-checkout.yml
+        - script: |
+            set -xe
+            export NO_CACHE=true
+            make azext-aro
 
-      - task: PublishCodeCoverageResults@1
-        displayName: 📈 Publish code coverage
-        inputs:
-          codeCoverageTool: Cobertura
-          summaryFileLocation: $(System.DefaultWorkingDirectory)/**/coverage.xml
-          failIfCoverageEmpty: false
-        condition: succeededOrFailed()
-        target: golang
+    - job: Lint_Az_ARO_Extension
+      pool:
+        name: 1es-aro-ci-pool
+      steps:
+        - template: ./templates/template-checkout.yml
+        - script: |
+            set -xe
+            make test-python
 
-  - job: Lint_Admin_Portal
-    pool:
-      name: 1es-aro-ci-pool
-    steps:
-      - template: ./templates/template-checkout.yml
-      - script: |
-          set -xe
-          make lint-admin-portal
-        displayName: 🧹 Lint Admin Portal
+- stage: Golang_Code_Generation
```

Comment:
nit: Can we add a #TODO on this to drop this stage once CI/CD is in place and committed generated artifacts no longer matter?

Reply:
Added a [Jira story ](https://issues.redhat.com/browse/ARO-9748)to track removing the Golang Code Generation stage once CI/CD is fully implemented.